### PR TITLE
Adds milestone progress column to dashboard

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/9277
fixes https://github.com/cityofaustin/atd-data-tech/issues/9349

## Testing
**URL to test:** 
https://deploy-preview-702--atd-moped-main.netlify.app/

**Steps to test:**

Go to dashboard, follow and add yourself to the team of a project if there is no project displayed under either tab
Add a project milestone to one of the projects and set it to incomplete, revisit dashboard
Add another project milestone to one of the projects and set it to complete, revisit dashboard and see the milestone column change

Follow a project
Add yourself as a team member of a project
Go to the dashboad
Click betweeb the "following" and "my projects" tabs and note that the table does not rerender when you switch tabs

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/dashboard/DashboardView.js [hunk 1]
```diff
@@ -186,34 +189,68 @@ const DashboardView = () => {
...
       editable: "never",
-      cellStyle: { ...typographyStyle, minWidth: "200px" },
+      cellStyle: { ...typographyStyle },
       render: entry => (
         <RenderFieldLink
           projectId={entry.project_id}
           value={entry.project_name}
         />
       ),
+      width: "25%",
```
Comment (tillyw) [inline moped-editor/src/views/dashboard/DashboardView.js:199]: Went ahead and updated the column widths to reflect the changes @chiaberry made in the milestones table

**Comment by tillyw** [comment]
In testing this I discovered an unrelated bug where "on hold" projects are not displayed in the "status" column (same issue as projects view). I tried fixing this but it's taking longer than expected so I'm suggesting we create a separate issue

**Review by amenity — APPROVED** [review]
😻 ❗ Wowwww. So sweet. Thanks, @tillyw !

**Review by mateoclarke — APPROVED** [review]
Great job Tilly! The percent counter looks perfect and the tab switch reload issue is fixed too!

clapping-bravo-stranger-things

**Review by chiaberry — APPROVED** [review]

**Comment by mateoclarke** [comment]
Since Tilly is out today and Monday, I'm going to merge.

**Pull request merged**